torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -28,31 +33,33 @@ ncclComm_t NCCLComm::getNcclComm() {
   }
   // only wait for initialization if nonblocking mode is enabled
   if (nccl_use_nonblocking()) {
```

Comment:
I think it's worth checking `initialized_` here to avoid calling multiple functions on the happy (and most common) path.  This skips two `nccl_` calls as well as `steady_clock::now()`.  Overhead would be small but IMO a bit cleaner.